### ISSUE TITLE
Name on iOS 14 does not include file extension 

### ISSUE THREAD

**Issue by aaahrens** [miguelpruivo/flutter_file_picker#444]
**Describe the bug**
When picking a file on iOS 14 the name does not include the extension. 

**Issue details**
1. Platform (Android/iOS) => iOS
2. Platform OS version => iOS 14
3. File type (filter) — when applicable
4. Detailed steps to reproduce=> Pick a file using the default example. 

**Flutter Version details**
`
[✓] Flutter (Channel dev, 1.23.0-4.0.pre, on Mac OS X 10.15.6 19G2021 x86_64, locale en-US)
    • Flutter version 1.23.0-4.0.pre at /Users/drunkengranite/flutter
    • Framework revision 83dd176777 (3 weeks ago), 2020-09-22 12:04:44 -0700
    • Engine revision 2abe69c608
    • Dart version 2.10.0 (build 2.10.0-136.0.dev)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/drunkengranite/Library/Android/sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Users/drunkengranite/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/193.6626763/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.9.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Users/drunkengranite/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/193.6626763/Android Studio.app/Contents
    • Flutter plugin version 48.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] IntelliJ IDEA Ultimate Edition (version 2020.2)
    • IntelliJ at /Users/drunkengranite/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin version 49.0.4
    • Dart plugin version 202.6397.47

[✓] VS Code (version 1.49.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.13.2

[✓] Connected device (4 available)
    • iPhone 11 Pro (mobile) • A6E0D885-D509-48A2-B9DC-97236C8EDCDE • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)
    • macOS (desktop)        • macos                                • darwin-x64     • Mac OS X 10.15.6 19G2021 x86_64
    • Web Server (web)       • web-server                           • web-javascript • Flutter Tools
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 86.0.4240.75

• No issues found!
`
**Additional context**
Would you guys be interested in setting up multiplatform testing?

**Comment by miguelpruivo** [comment]
@aaahrens tried to replicate this without success. File always shows its extension. Could you please elaborate how to replicate it?

Thank you!

**Comment by miguelpruivo** [comment]
@aaahrens any news regarding this?

**Comment by aaahrens** [comment]
Um yeah I'll make an example repo with steps and video proof, but it's already confirmed by multiple devs on my team.  Just to be clear, it's on iOS 14, FileType.any and with flutter channel master.

**Comment by miguelpruivo** [comment]
@aaahrens the plugin is only tested against stable channel as master has a lot under development. I couldn’t reproduce it with stable, could that be the issue?

**Comment by orestesgaolin** [comment]
On current stable 1.22.2, the same issue on iOS simulator running iOS 14.0

```
          final result = await FilePicker.platform.pickFiles(
            type: FileType.custom,
            allowedExtensions: ['jpg', 'png', 'pdf'],
          );
```

Download some PDF to your phone's files. Go to Browse > On my iPhone > Downloads > select some PDF file 

![image](https://user-images.githubusercontent.com/16854239/96693934-1b557f00-1388-11eb-8025-dcf65007a419.png)


The resulting `result.files.single.name` is `sample` and `result.files.single.path` is `/Users/dominik/Library/Developer/CoreSimulator/Devices/CA2A19B0-A363-4D5A-9C59-D523D30124C1/data/Containers/Data/Application/29343E05-E76D-4A9F-922B-C0B11E83DE04/tmp/com.app/sample.pdf`

**Comment by miguelpruivo** [comment]
This is quite odd as the extension is extracted from the file name itself (it’s just a helper getter anyway).

I’ll take a look ASAP. Thank you.

**Comment by orestesgaolin** [comment]
I wonder if the file extension is removed in [this line](https://github.com/miguelpruivo/flutter_file_picker/blob/master/ios/Classes/FileUtils.m#L92)

**Comment by miguelpruivo** [comment]
@orestesgaolin that's correct. I think that was probably because at first I was considering not including the extension in the `name` property but then changed my mind later on and missed that. 😅 

Thank you all for reporting.

Fixed and published with [2.0.10](https://pub.dev/packages/file_picker/changelog#2010).
